pandas/core/arrays/datetimes.py

```diff
@@ -1918,6 +1918,9 @@ def sequence_to_dt64ns(
     tz = validate_tz_from_dtype(dtype, tz)
 
     if isinstance(data, ABCIndexClass):
+        if data.nlevels > 1:
+            # Without this check, data._data below is None
+            raise TypeError("Cannot create a DatetimeArray from a MultiIndex.")
```

Comment:
we should have an index method which does this no?

Reply:
For the check or the raising?  The check could be `isinstance(data, ABCMultiIndex)`.  No strong preference.

Reply:
no, something that does the right thing on an Index/MultiIndex, e.g. that gets (data._data) on an index and just passes thru on a MI

Reply:
I think we want to move to use the `extract_array` pattern here, but that currently raises ValueError on MultiIndex (where we want to raise TypeError here)

Reply:
ok the reason I am harping on this is its an inconsistency between Index and MultiIndex. ok on merging this and creating an issue / followup ?

Reply:
> creating an issue / followup ?

yes